aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -7,81 +7,295 @@
+            if not src.exists():
+                raise exceptions.ArchiveExportError(
+                    f'Unable to find the repository folder for Node with UUID={uuid} '
+                    'in the local repository'
+                )
+
+            # In this way, I copy the content of the folder, and not the folder itself
+            thisnodefolder.insert_path(src=src.abspath, dest_name='.')
+
+
+class WriterJsonZip(WriterAbstract):
+
+    def __init__(self, filename: str, **kwargs: Any):
+        """An archive writer
+
+        :param filename: the filename (possibly including the absolute path)
+            of the file on which to export.
+        :param use_compression: Whether or not to compress the zip file.
+
+        """
+        super().__init__(filename, **kwargs)
+        self._use_compression = kwargs.get('use_compression', True)
+
+    @property
+    def file_format_verbose(self) -> str:
+        return 'Zip (compressed)' if self._use_compression else 'Zip (uncompressed)'
+
+    def write(
+        self,
+        export_data: ExportData,
+        silent: bool = False,
+    ) -> dict:
+        """write the archive
+
+        :param entities: a list of entity instances; they can belong to different models/entities.
+        :param traversal_rules: graph traversal rules. See :const:`aiida.common.links.GraphTraversalRules`
+            what rule names are toggleable and what the defaults are.
+
+        """
+        with ZipFolder(self.filename, mode='w', use_compression=self._use_compression) as folder:
+            _write_to_json_archive(folder=folder, export_data=export_data, silent=silent)
+
+        return {}
+
+
+class WriterJsonTar(WriterAbstract):
+
+    @property
+    def file_format_verbose(self) -> str:
+        return 'Gzipped tarball (compressed)'
+
+    def write(
+        self,
+        export_data: ExportData,
+        silent: bool = False,
+    ) -> dict:
+        """write the archive
+
+        :param entities: a list of entity instances; they can belong to different models/entities.
+        :param traversal_rules: graph traversal rules. See :const:`aiida.common.links.GraphTraversalRules`
+            what rule names are toggleable and what the defaults are.
+
+        """
+        with SandboxFolder() as folder:
+            _write_to_json_archive(folder=folder, export_data=export_data, silent=silent)
+
+            with tarfile.open(self.filename, 'w:gz', format=tarfile.PAX_FORMAT, dereference=True) as tar:
+                time_compress_start = time.time()
+                tar.add(folder.abspath, arcname='')
+                time_compress_stop = time.time()
+
+        return {'compression_time_start': time_compress_start, 'compression_time_stop': time_compress_stop}
+
+
+def get_writers() -> Dict[str, Type[WriterAbstract]]:
+    """Return the available writer classes."""
+    # TODO this could be made an entrypoint
```

Comment:
how do you feel about making it an entry point in this PR?

Reply:
Adding new entry point groups should be discussed among more than the two of us, so I suggest you open an issue for this now with a proposition for the new entry point group.

If consensus happens to be reached before this PR is merged, we can include it here - otherwise let's go ahead without.